src/controlobjectscript.cpp

```diff
@@ -0,0 +1,45 @@
+#include <QApplication>
+#include <QtDebug>
+
+#include "controlobjectscript.h"
+
+ControlObjectScript::ControlObjectScript(const ConfigKey& key, QObject* pParent)
+        : ControlObjectSlave(key, pParent) {
+}
+
+ControlObjectScript::~ControlObjectScript() {
+    //qDebug() << "ControlObjectScript::~ControlObjectSlave()";
+}
+
+bool ControlObjectScript::connectScriptFunction(
+        const ControllerEngineConnection& conn) {
+    m_connectedScriptFunction.append(conn);
+    connect(m_pControl.data(), SIGNAL(valueChanged(double, QObject*)),
+            this, SLOT(slotValueChanged(double,QObject*)),
+            static_cast<Qt::ConnectionType>(Qt::QueuedConnection |
+                                            Qt::UniqueConnection));
+    connect(this, SIGNAL(trigger(double, QObject*)),
+            this, SLOT(slotValueChanged(double,QObject*)),
+            static_cast<Qt::ConnectionType>(Qt::QueuedConnection |
+                                            Qt::UniqueConnection));
+    return true;
+}
+
+bool ControlObjectScript::disconnectScriptFunction(
+        const ControllerEngineConnection& conn) {
+    return m_connectedScriptFunction.removeAll(conn) > 0;
+}
+
+void ControlObjectScript::slotValueChanged(double value, QObject*) {
+    foreach(ControllerEngineConnection conn, m_connectedScriptFunction) {
```

Comment:
I think you can use a standard C++11 range-based for loop to iterate through QList. We should try to avoid the Qt hack for newly written code.